clang/runtime/dpct-rt/include/dpct/sparse_utils.hpp

```diff
@@ -830,6 +832,57 @@ inline void csrsv(sycl::queue &queue, oneapi::mkl::transpose trans, int row_col,
                                           optimize_info, x, y);
 }
 
+/// Performs internal optimizations for dpct::sparse::csrsm by analyzing
+/// the provided matrix structure and operation parameters.
+/// \param [in] queue The queue where the routine should be executed. It must
+/// have the in_order property when using the USM mode.
+/// \param [in] transa The operation applied to A.
+/// \param [in] transb The operation applied to B and X.
+/// \param [in] row_col Number of rows and columns of A.
+/// \param [in] nrhs Number of columns op_b(B).
+/// \param [in] info Matrix info of A.
+/// \param [in] val An array containing the non-zero elements of A.
+/// \param [in] row_ptr An array of length \p num_rows + 1.
+/// \param [in] col_ind An array containing the column indices in index-based
+/// numbering.
+/// \param [out] optimize_info The result of the optimizations.
+template <typename T>
+void optimize_csrsm(sycl::queue &queue, oneapi::mkl::transpose transa,
+                    oneapi::mkl::transpose transb, int row_col, int nrhs,
+                    const std::shared_ptr<matrix_info> info, const T *val,
+                    const int *row_ptr, const int *col_ind,
+                    std::shared_ptr<optimize_info> optimize_info) {
+  detail::optimize_csrsm_impl<T>()(queue, transa, transb, row_col, nrhs, info,
+                                   val, row_ptr, col_ind, optimize_info);
+}
+
+/// Solves the sparse triangular system op_a(A) * op_b(X) = alpha * op_b(B)
```

Comment:
1. please explict tell which is out in the formula.  2.  highlight the data format required.